lib/private/DB/Connection.php

```diff
@@ -527,6 +529,11 @@ public function lastInsertId($name = null): int {
 	 * @throws Exception
 	 */
 	public function realLastInsertId($seqName = null) {
+		if ($this->lastInsertId !== null && $this->lastInsertId !== '0') {
```

Comment:
I hate the type juggling on lastInsertId, it should be strong typed to ?int instead.
```suggestion
		if ($this->lastInsertId !== null && $this->lastInsertId !== 0) {
```